src/norm_flows.jl

```diff
@@ -0,0 +1,88 @@
+using Distributions
+using LinearAlgebra
+using Random
+
+################################################################################
+#                            Planar and Radial Flows                           #
+#             Ref: Variational Inference with Normalizing Flows,               #
+#               D. Rezende, S. Mohamed(2015) arXiv:1505.05770                  #
+################################################################################
+
+mutable struct PlanarLayer <: Bijector
+    w
+
+    u
+    u_hat
+    b
+end
+
+mutable struct RadialLayer <: Bijector
+    α_
+    β
+    z_not
+end
+
+function update_u_hat(u, w)
+    # to preserve invertibility
+    u_hat = u + (m(transpose(w)*u) - transpose(w)*u)[1]*w/(norm(w[:,1],2)^2)
+end
+
+function update_u_hat!(flow::PlanarLayer)
+    flow.u_hat = flow.u + (m(transpose(flow.w)*flow.u) - transpose(flow.w)*flow.u)[1]*flow.w/(norm(flow.w,2)^2)
+end
+
+
+function PlanarLayer(dims::Int)
+    w = param(randn(dims, 1))
+    u = param(randn(dims, 1))
+    b = param(randn(1))
+    u_hat = update_u_hat(u, w)
+    return PlanarLayer(w, u, u_hat, b)
+end
+
+function RadialLayer(dims::Int)
+    α_ = param(randn(1))
+    β = param(randn(1))
+    z_not = param(randn(dims, 1))
+    return RadialLayer(α_, β, z_not)
+end
+
+m(x) = -1 .+ log.(1 .+ exp.(x)) #for planar flow
+dtanh(x) = 1 .- (tanh.(x)).^2 #for planar flow
+ψ(z, w, b) = dtanh(transpose(w)*z .+ b).*w #for planar flow
+softplus(x) = log.(1 .+ exp.(x)) #for radial flow
```

Comment:
Please use the one from StatsFuns.jl

Reply:
The one from StatsFuns isn't able to handle tracked input.
```
MethodError: no method matching log1pexp(::TrackedArray{…,Array{Float64,1}})
Closest candidates are:
  log1pexp(!Matched::Float32) at /home/sharan/.julia/packages/StatsFuns/2QE7p/src/basicfuns.jl:67
  log1pexp(!Matched::Real) at /home/sharan/.julia/packages/StatsFuns/2QE7p/src/basicfuns.jl:66
```

Reply:
Likely you need to use `log1pexp.` instead of `log1pexp`